R/sfg.R

```diff
@@ -26,14 +25,17 @@ MtrxSet = function(x, dim = "XYZ", type, needClosed = FALSE) {
 	if (length(x) > 0) { # list()
 		nc = unique(vapply(x, ncol, 0L))
 		if (length(nc) != 1)
-			stop("matrices having unequal number of columns")
+			stop("matrices have unequal numbers of columns")
 		lapply(x, is_numeric_matrix)
-		NotClosed = function(y) any(head(y, 1) != tail(y, 1))
+		NotClosed = function(y) any(y[1, ] != y[nrow(y), ])
 		if (needClosed && any(vapply(x, NotClosed, TRUE)))
 			stop("polygons not (all) closed")
-		structure(x, class = getClassDim(x, nc, dim, type))
-	} else
-		structure(x, class = getClassDim(x, nchar(dim), dim, type))
+		class(x) = getClassDim(x, nc, dim, type)
+		return(x)
```

Comment:
I can't believe that this brings improvement; I can understand you find this more readable, but since the rest of `sf` also uses `structure` and avoids returns()s, please leave this the way it was.

Reply:
I made the change for performance. Using `structure` instead of `class` increases `st_make_grid` runtime by 15-30%, which isn't that surprising if you look at the implementation of `structure`. Obviously you make the call on the performance/style tradeoff.

Reply:
OK, that's a good reason.